perceval/backends/core/zulip.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Venu Vardhan Reddy Tekula <venuvardhanreddytekula8@gmail.com>
+#
+
+
+import logging
+
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+logger = logging.getLogger(__name__)
+
+CATEGORY_MESSAGE = "message"
+
+
+class Zulip(Backend):
+    """Zulip Backend."""
+
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    def __init__(self, url, stream, api_token, email, origin, tag=None, archive=None, ssl_verify=True):
```

Comment:
Origin shouldn't be declared in __init__, since it can be derived from the url and stream. The origin is used to identify a repository.
Line 45 and 48 are duplicated.
The url should be stored a instance attribute, since it will be passed then to the client

Suggested changes below:
```
-    def __init__(self, url, stream, api_token, email, origin, tag=None, archive=None, ssl_verify=True):
-        super().__init__(origin, tag, archive, ssl_verify)
-        origin = url
-
-        super.__init__(origin, tag=tag, archive=archive, ssl_verify=True)
+    def __init__(self, url, stream, api_token, email, tag=None, archive=None, ssl_verify=True):
+        origin = urijoin(url, stream)
+        super().__init__(origin, tag=tag, archive=archive, ssl_verify=ssl_verify)
+        self.url = url
         self.stream = stream
         self.api_token = api_token
         self.email = email
 
         self.client = None
```

Reply:
Thanks, I was a bit confused about this. I got an idea about it now. :smile: 

Also, I had one more doubt about the `origin`. 
The original URL of the stream is https://python.zulipchat.com/#narrow/stream/187177-importlib.
The `origin` which we generates here is `https://python.zulipchat.com/importlib`.

I was just having second thoughts about this now.

Reply:
> Thanks, I was a bit confused about this. I got an idea about it now. smile

You're welcome!

I see that `https://python.zulipchat.com/#narrow/stream/importlib` is equivalent to https://python.zulipchat.com/#narrow/stream/187177-importlib. We could set the origin with the first option, which is easier to build, WDYT?

Reply:
Okay, perfect. :+1: